### PR TITLE
feat: Improve Terraform registry GitHub integration

### LINKED ISSUE #5: Throw 404s when the repositories do not exist
This is to be in keeping with other registries

### PR DESCRIPTION
**Pull request by Yantrio**
Fixes #5 

- Add function to check if a GitHub repository exists using `go-github`.
- Refactor functions to use constructed repo names for querying releases.
- Implement repo existence check in Lambda functions before processing.
- Simplify repo name construction by removing unnecessary string formatting.
- Add a centralized 404 Not Found response for easier error handling.

### CODE REVIEW

File: lambda/moduleDownload.go [hunk 1]
```diff
@@ -17,12 +18,21 @@ func downloadModuleVersion(config Config) LambdaFunc {
 	return func(ctx context.Context, req events.APIGatewayProxyRequest) (events.APIGatewayProxyResponse, error) {
 		params := getDownloadModuleHandlerPathParams(req)
 
-		url := modules.GetVersionDownloadUrl(ctx, params.Namespace, params.Name, params.System, params.Version)
+		// the repo name should match the format `terraform-<system>-<name>`
+		repoName := fmt.Sprintf("terraform-%s-%s", params.System, params.Name)
```
Comment (RLRabinowitz) [inline lambda/moduleDownload.go]: nit
Personally I'll create a common for constructing the repository name from the module/provider name

**Review by RLRabinowitz — COMMENTED** [review]
Looks good, left one small suggestion

File: lambda/moduleDownload.go (hunk 1 above)
Comment (Yantrio) [reply lambda/moduleDownload.go]: Good call. I'll do that quickly now

Comment (Yantrio) [reply lambda/moduleDownload.go]: Resolved in https://github.com/opentffoundation/registry/pull/12/commits/1a8b9e80564bf47dac737db8e739f51036879614

**Review by marcinwyszynski — APPROVED** [review]

**Pull request merged**